### DISCUSSION TITLE
How can one achieve a "Partial-View" approach using liquid views?

### DISCUSSION THREAD

**Discussion by SzymonSel** [OrchardCMS/OrchardCore#14401] [Q&A]
So I've go a use case, where our `view` is huge and has some reuseble parts. What is the approach using `ContentItems`, `ContentParts`, `Shapes` etc. to achieve this goal?

**Comment by hishamco** [comment]
Have a look to https://github.com/sebastienros/fluid/tree/main/Fluid.MvcSample/Views

**Reply by SzymonSel** [reply]
Ahh, I see. Is the `invoker` object is by convention the "model" for the partial view or does the parent share all it's variables and ViewModel with the partial view?

**Reply by SzymonSel** [reply]
The example code uses `Fluid.MvcViewEngine` and needs `services.AddMvc().AddFluid()`, which somewhat is in conflict with `services.AddOrchardCms()`. How can I fix this?

**Comment by ns8482e** [comment]
Orchard core has concept of Shapes. Orchard core's display management builds content item's display using different shapes.  Rendering of these shapes can be changed by  defining templates, wrappers, alternates.  Liquid and Razor both templates are supported.

See [documentation for templates](https://docs.orchardcore.net/en/main/docs/reference/modules/Templates/) 

If you are specifically looking for rendering a shape from another shape in liquid then, you can also use [shape filters](https://docs.orchardcore.net/en/main/docs/reference/modules/Liquid/#shape-filters)
